### DISCUSSION TITLE
Set number of crawler manually

### DISCUSSION THREAD

**Discussion by marctuerke** [eclipse-edc/FederatedCatalog#268] [General]
When I set the number of crawler in the deployment of my catalog server, the deployment is invalid. It says, that Kubernetes cannot unmarshal the Env Value of the number of crawlers from a number to type string. The error persists even when I change the data type of the number of crawlers inside the values.yaml.

**Comment by paullatzelsperger** [comment]
@marctuerke I have no way of knowing how you are deploying FC to your K8S cluster, but I recommend to checkout the [MVD](https://github.com/eclipse-edc/MinimumViableDataspace/) project, where we use Terraform, and configuration is mapped into a `ConfigMap`. I just verified that it is indeed possible to set the number of crawlers to 10.

That said, the _actual_ number of crawlers that are spun up will also depend on the number of work items in your queue. Meaning, if your `TargetNodeDirectory` only contains 5 elements, but your `edc.catalog.cache.partition.num.crawlers=10`, then the FC will still only spin up 5 crawlers.

**Comment by marctuerke** [comment]
Sorry forgot to mention, I use the helm chart of the TractusX-EDC and when I set the number of crawlers there, the controlplane can not be deployed and throws the following error:

```
Failed sync attempt to : one or more objects failed to apply, reason: 
Deployment in version "v1" cannot be handled as a Deployment: 
json: cannot unmarshal number into Go struct field EnvVar.spec.template.spec.containers.env.value of type string (retried 5 times).
```

When I do not set the number of crawlers it works.

**Reply by paullatzelsperger** [reply]
then this is the wrong repository to ask this question, because it clearly has to do with the Tractus-X EDC Helm chart rather than the federated catalog itself. This error is not coming from Kubernetes but from Helm I think. 
I'll close this question here, but feel free to ask it again in the Tractus-X EDC repo.
